### DISCUSSION TITLE
Viessman ViCare - setting occupied_heating_setpoint doesn't work

### DISCUSSION THREAD

**Discussion by a96965974b61** [Koenkk/zigbee2mqtt#11066] [Devices]
Hi,

I have ViCare ZK03840 TRV (software reported as 01.16.0008 02.04) paired with zigbee2mqtt (1.22.2). It works partially. 

So, for example:
- reading local_temperature works;
- changing Keypad_lockout or system_mode works;

BUT when I'm setting occupied_heating_setpoint then value on display changes, but nothing happens.

At the same time if I go with the following sequence:
- set system_mode to sleep -> action is triggered;
- set occupied_heating_setpoint -> value on display changed;
- set system_mode to heat -> action is triggered.

Am I missing something? What is the correct way of changing expected temperature?

**Comment by sjorge** [comment]
Was the system_mode still set to `sleep` when you set occupied_heating_setpoint ? If so it's probably not doing anything because the mode is still `sleep`

**Comment by a96965974b61** [comment]
No, for sure system mode was set to **heat**. Nevertheless, I need to do more checks. Basic setup is like:

1. local_temperature  15C
2. if radiator is manually set to 8C, it closes
3. if radiator is manually set to 22C, it opens
4. if radiator is set to 8C via setting occupied_heating_setpoint (while at 22C), nothing happens
5. if radiator is set to 22C via setting occupied_heating_setpoint (while at 8C), nothing happens

However, I need to validate points 4 & 5 in more details. It seems nothing happens directly after setting, but after long time the setting is actually adjusted. The "long time" can be more than 15 minutes. Is such delay in reaction normal?

**Reply by sjorge** [reply]
That might be normal, the battery powered devices sleep a lot to save power. So it might not check in for new message more than once every 15sec or longer. But if you adjust it physically it wakes up.

**Comment by a96965974b61** [comment]
I'm not talking about seconds. So, test from today:
- temperature set to 22 at 6 AM
- NOTHING happens till 8 AM, temperature reported by local_temperature still drops (it was set to 19 in the evening)
- At 8:40 system_mode is set to heat (it was set to heat before, I re-checked it before setting it)
- at this point temperature starts increasing towards 22.

Well, it seems I will just modify thing description in OpenHab to send occupied_heating_setpoint with system_mode: heat
